src/test_s4solution.cpp

```diff
@@ -288,6 +359,27 @@ class SimpleS4SolutionTest : public BaseS4SolutionTest
   }
 };
 
+/// Add a key let it time out.
+TEST_F(SimpleS4SolutionTest, TimerTracerBullet)
+{
+  const std::string impu = "sip:kermit@muppets.com";
+
+  // PUT a binding to site 1.
+  AoR* aor = new AoR(impu);
+  Binding* b = new Binding(impu);
+  b->_expires = time(nullptr) + 3;
+  aor->_bindings[impu] = b;
+
+  _s4_site1->s4->handle_put(impu, *aor, FAKE_SAS_TRAIL_ID);
+  delete aor; aor = nullptr;
+
+  EXPECT_CALL(_s4_site1->timer_sink, handle_timer_pop(impu, _));
+  sleep(5);
+
+  printf("Test complete\n");
+  sleep(3600);
```

Comment:
So I checked out this branch, and tried to run the two S4 tests. Here's what happened...
* I had to take more up to date versions of clearwater-s4 (`a7e373b...`) and cpp-common (`f1f51135331d2658683071fb08b4bb762c710bbf`) than this branch points at (fine, but worth noting in case that's unexpected for some reason!)
* I had to add `${ROOT}/usr/lib` to the `LD_LIBRARY_PATH` in the `run_test` target in `src/Makefile`. Without it, I got the following error when running `make run_test`:

```
LD_LIBRARY_PATH=/home/ubuntu/clearwater-fv-test/modules/astaire/usr/lib /home/ubuntu/clearwater-fv-test/build/bin/fvtest  --gtest_output=xml:/home/ubuntu/clearwater-fv-test/build/testout/test_detail_fvtest.xml
/home/ubuntu/clearwater-fv-test/build/bin/fvtest: error while loading shared libraries: libevent-2.1.so.3: cannot open shared object file: No such file or directory
make: *** [run_test] Error 127
```

This is inexplicable to me
* I then deleted these two lines (because they didn't look right?!) and ran `make run_test JUSTTEST="SimpleS4SolutionTest.*"`. The second test failed; is that expected? It fails in isolation too.

```
[ RUN      ] SimpleS4SolutionTest.TracerBullet
test_s4solution.cpp:405: Failure
Value of: HTTP_OK
  Actual: 200
Expected: status
Which is: 500
test_s4solution.cpp:406: Failure
Expected: (aor) != (nullptr), actual: NULL vs 8-byte object <00-00 00-00 00-00 00-00>
[  FAILED  ] SimpleS4SolutionTest.TracerBullet (5732 ms)
```

Reply:
I've fixed up the submodule refs (I did know they were wrong but I was just going to leave it for the master builds to fix up). I've also removed the two rogue lines. I don't understand why you needed the `LD_LIBRARY_PATH` stuff of why the test then didn't work. It obviously works on my box. I am going to find another server to run it on to dig into this.

Reply:
Has `libevent-2.1.so.3` made it into your `LD_LIBRARY_PATH` via some other system-wide means? Not 100% sure what I'm talking about.

Reply:
That'll be it. Libevhtp depends on libevent which I have installed because I've built production code in my box. I'll update the README.

Reply:
I'm so hot right now 🔥 🔥 🔥

Reply:
Spoke too soon, I've installed libevent-dev as a dependency but I still have the same error. Where is your `libevent-2.1.so.3`?

```
ubuntu@gdr-dev~/clearwater-fv-test/src (sdm-timer-pops) $ locate libevent-2.1.so.3
/home/ubuntu/clearwater-fv-test/usr/lib/libevent-2.1.so.3
/home/ubuntu/clearwater-fv-test/usr/lib/libevent-2.1.so.3.0.0
```

Reply:
As discussed, that version of `libevent` I had was rogue and the tests now both run and pass

Reply:
@AlexHockey Have you run `make vg_raw` successfully?

Prediction: no